sql-plugin/src/main/scala/com/nvidia/spark/rapids/GpuOrcScan.scala

```diff
@@ -858,6 +897,74 @@ trait OrcCommonFunctions extends OrcCodecWritingHelper { self: FilePartitionRead
         throw new IOException(s"Error when processing file splits [${splits.mkString("; ")}]", e)
     }
   }
+
+  protected final def isNeedToSplitDataBlock(
+      curMeta: OrcBlockMetaForSplitCheck,
+      nextMeta: OrcBlockMetaForSplitCheck): Boolean = {
+    if (!nextMeta.typeDescription.equals(curMeta.typeDescription)) {
+      logInfo(s"ORC schema for the next file ${nextMeta.filePath}" +
+        s" schema ${nextMeta.typeDescription} doesn't match current ${curMeta.filePath}" +
+        s" schema ${curMeta.typeDescription}, splitting it into another batch!")
+      return true
+    }
+
+    if (nextMeta.compressionKind != curMeta.compressionKind) {
+      logInfo(s"ORC File compression for the next file ${nextMeta.filePath}" +
+        s" doesn't match current ${curMeta.filePath}, splitting it into another batch!")
+      return true
+    }
+
+    val ret = (nextMeta.requestedMapping, curMeta.requestedMapping) match {
+      case (None, None) => true
+      case (Some(cols1), Some(cols2)) =>
+        if (cols1.sameElements(cols2)) true else false
+      case (_, _) => {
+        false
+      }
+    }
+
+    if (!ret) {
+      logInfo(s"ORC requested column ids for the next file ${nextMeta.filePath}" +
+        s" doesn't match current ${curMeta.filePath}, splitting it into another batch!")
+      return true
+    }
+
+    false
+  }
+
+  protected implicit def toStripe(block: DataBlockBase): OrcStripeWithMeta =
+    block.asInstanceOf[OrcDataStripe].stripeMeta
+
+  protected implicit def toDataStripes(stripes: Seq[DataBlockBase]): Seq[OrcStripeWithMeta] =
+    stripes.map(_.asInstanceOf[OrcDataStripe].stripeMeta)
+
+  protected final def estimateOutputSizeFromBlocks(blocks: Seq[OrcStripeWithMeta]): Long = {
+    // Start with header magic
+    var size: Long = OrcFile.MAGIC.length
```

Comment:
personally prefer not to see var's (mutable datastructures) used unless really needed or makes things that much easier, just use vals and add them up at the end.

Reply:
done